### PR TITLE
chore(a11y): update accessibility-checker to v3.1.74

### PR DESCRIPTION
**Pull request by tay1orjones**
This PR updates accessibility-checker to the latest.

I'm not sure why renovate hasn't been keeping this up to date. @guidari any ideas? 🤔 

#### Changelog

**Changed**

- update accessibility-checker to latest
- update our ruleset config

#### Testing / Reviewing

- There's quite a few new a11y failures from the [latest `06 August 2024`](https://www.ibm.com/able/requirements/checker-rule-sets) ruleset here that we need to fix in order to merge this

### CODE REVIEW

File: achecker.js [hunk 1]
```diff
@@ -10,7 +10,7 @@
 const path = require('path');
 
 module.exports = {
-  ruleArchive: 'latest',
+  ruleArchive: 'versioned',
```
Comment (tay1orjones) [inline achecker.js]: This makes sure we use a rule pack that matches the version of `accessibility-checker` we're currently on. New rulesets will only be brought in by updating `accessibility-checker`, so no more randomly failing CI checks out of the blue after a new ruleset is published on `latest`.

**Review by 2nikhiltom — APPROVED** [review]

**Review by guidari — APPROVED** [review]
> I'm not sure why renovate hasn't been keeping this up to date. @guidari any ideas? 🤔

Idk. I'm gonna dig to understand why.

**Comment by guidari** [comment]
Here it is why @tay1orjones 
https://github.com/carbon-design-system/carbon/pull/17133

**Review by emyarod — APPROVED** [review]
looks good to me! just the one conflict in `achecker` needs to be resolved

**Pull request merged**